_code/InSituDiff/vignettes/InSituDiff basic vignette.Rmd

```diff
@@ -0,0 +1,220 @@
+ one with childhood lupus and two controls:
+
+```{r prep}
+library(InSituDiff)
+library(pheatmap)
+# load demo data from lupus nephritis:
+data(minisle)
+# normalize the counts matrix (InSituDiff is best run on normalized data):
+normconstants <- pmax(Matrix::rowSums(minisle$counts), 20) 
+norm_factors <- Matrix::Diagonal(x = 1 / normconstants)
+minisle$norm <- norm_factors %*% minisle$counts * mean(normconstants)
+rownames(minisle$norm) <- rownames(minisle$counts)
+```
+
+## Setting up InSituDiff
+
+Now we set ourselves up for calculating perturbation scores by running ```initializeISD```.
+This function defines two mappings needed for perturbation scoring: 
+
+1. The cells making up each cell's neighborhood
+2. For each cell, the identify of the control well with the most-similar neighborhood
+
+```{r initialize}
+obj <- initializeISD(
+  mat = minisle$norm, # matrix of normalized expression
+  xy = minisle$xy, # matrix of xy coordinates
+  tissue = minisle$annot$tissue, # vector of cells' tissue IDs
+  iscontrol = grepl("control", minisle$annot$tissue), 
+  k = 50) # define cellular neighborhoods as a cell's 50 nearest neighbors
+str(obj)
+```
+
+And that's all we need to start analysis.  
+
+## Calculating neighborhood perturbations
+
+Below we'll show three useful analyses of perturbations we can perform. 
+But first, at any time you like, you can easily compute a matrix of perturbation scores
+ for any cells and genes you like. 
+To get perturbations for all cells and genes, just enter ```cells = NULL, genes = NULL```.
+*Warning*: for larger datasets, attempting to compute this matrix for all cells and all genes risks overwhelming your R session's memory.
+
+```{r getdata}
+selectedperturbations <- getPerturbations(
+  x = minisle$norm, 
+  obj = obj, 
+  cells = rownames(minisle$counts)[minisle$annot$tissue == "SP19_1139"], # only look at one tissue
+  genes = c("IGHA1", "FGFR3", "HIF1A", "ACE2", "SAT1"),  # only look at a few genes of interest
+  residtype = "log2ratio",  # look at perturbations on the scale of log fold-changes
+  eps = 1)
+# the result is just dense matrix of perturbation scores over cells x genes:
+selectedperturbations[1:5,1:5]
+```
+
+## Identifying highly perturbed genes
+
+We summarize genes' degree of perturbation in two ways: how broadly perturbed they are, 
+ and how intensely perturbed they are.
+We calculate "breadth" of perturbation by comparing a gene's medium perturbation score in disease vs. control. 
+We calculate "intensity" of perturbation by comparing a gene's mean+2SD perturbation score in disease vs. control. 
+
+Plotting these scores against each other gives a good sense of how genes are perturbed:
+
+```{r genepertlevels}
+genescores <- summarizeGenePerturbation(
+  x = minisle$norm, obj = obj, 
+  residtype = "log2ratio", plotresults = TRUE, eps = 0.1, subsetsize = 1e5) 
+```
+
+To get a feel for what these statistics mean, let's look at a broadly perturbed 
```

Comment:
Can you define which of these genes is broadly perturbed and which one is intensely perturbed? It's not clear to me from the text or the plots. Most genes fall along the same xy line - is this an anomaly because it's such a small dataset? If so, it might be worth mentioning that in larger datasets you do expect to see more genes falling off that line and thus being more broadly or intensely perturbed.